lib/scanner/walk.go

```diff
@@ -226,6 +239,8 @@ func (w *walker) walkAndHashFiles(ctx context.Context, fchan, dchan chan protoco
 
 		if !utf8.ValidString(path) {
 			l.Warnf("File name %q is not in UTF8 encoding; skipping.", path)
+			// Purposely do not send a ScanResult with an error, as this can never
+			// be recovered, as this invalid path can never be succesfully scanned.
```

Comment:
Not clear on the reasoning here. Isn't this similar to any other scan error like permission denied - something we flag and the user can fix?